src/domain/event/queryUtils.ts

```diff
@@ -0,0 +1,224 @@
+  getSearchQuery,
+} from '../eventSearch/utils';
+import { SIMILAR_EVENTS_AMOUNT } from './constants';
+import { getEventFields, getEventIdFromUrl } from './EventUtils';
+import { EventFields } from './types';
+
+const useSimilarEventsQueryVariables = (event: EventFields) => {
+  const locale = useLocale();
+  const { search } = useLocation();
+  const { keywords } = getEventFields(event, locale);
+  const eventSearch = getSearchQuery({
+    ...DEFAULT_SEARCH_FILTERS,
+    keyword: keywords.map((keyword) => keyword.id),
+  });
+
+  // Filter by search query if exists, if not filter by event keywords
+  const searchParams = new URLSearchParams(search || eventSearch);
+  return React.useMemo(() => {
+    return getEventSearchVariables({
+      include: ['keywords', 'location'],
+      language: locale,
+      pageSize: PAGE_SIZE,
+      params: searchParams,
+      sortOrder: EVENT_SORT_OPTIONS.END_TIME,
+      superEventType: ['umbrella', 'none'],
+    });
+  }, [locale, searchParams]);
+};
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const useSimilarEventsQuery = (event: EventFields) => {
+  const eventFilters = useSimilarEventsQueryVariables(event);
+
+  const { data: eventsData, loading } = useEventListQuery({
+    ssr: false,
+    variables: eventFilters,
+  });
+
+  // To display only certain amount of events.
+  // Always fetch data by using same page size to get events from cache
+  const data =
+    eventsData?.eventList.data
+      // Don't show current event on the list
+      .filter((item) => item.id !== event.id)
+      .slice(0, SIMILAR_EVENTS_AMOUNT) || [];
+
+  return { data, loading };
+};
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const useSimilarCoursesQuery = (event: EventFields) => {
+  const eventFilters = useSimilarEventsQueryVariables(event);
+
+  const { data: eventsData, loading } = useCourseListQuery({
+    ssr: false,
+    variables: eventFilters,
+  });
+
+  // To display only certain amount of events.
+  // Always fetch data by using same page size to get events from cache
+  const data =
+    eventsData?.courseList.data
+      // Don't show current event on the list
+      .filter((item) => item.id !== event.id)
+      .slice(0, SIMILAR_EVENTS_AMOUNT) || [];
+
+  return { data, loading };
+};
+
+const useOtherEventTimesVariables = (event: EventFields) => {
+  const superEventId = React.useMemo(
+    () => getEventIdFromUrl(event.superEvent?.internalId || ''),
+    [event.superEvent]
+  );
+
+  const variables = React.useMemo(
+    (): EventListQueryVariables => ({
+      include: ['keywords', 'location'],
+      sort: EVENT_SORT_OPTIONS.START_TIME,
+      start: 'now',
+      superEvent: superEventId,
+    }),
+    [superEventId]
+  );
+
+  return { superEventId, variables };
+};
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const useOtherEventTimes = (event: EventFields) => {
```

Comment:
Could this be combined with `useOtherCourseTimes` ? 
```suggestion
export const useOtherEventTimes = (event: EventFields, useQuery = useEventListQuery, listType = 'eventList') => { ... }
```

Reply:
With this I thought that the typing would be even harder, but maybe I need to try something to find out if it is doable with good typings

Reply:
```
export const useOtherEventTimes = (
  event: EventFields,
  useQuery: typeof useEventListQuery | typeof useCourseListQuery
) => {
```

Maybe something like that and some kind of type guarding will make this possible

Reply:
Or if you have other ideas to handle typing, please tell me :)